File: cpp_pkg/src/laser_sub.cpp
#include <memory>
#include "rclcpp/rclcpp.hpp"
#include "sensor_msgs/msg/laser_scan.hpp"

using LaserScan = sensor_msgs::msg::LaserScan;

class LaserSub : public rclcpp::Node
{
 public:
  LaserSub(const std::string& model);

  void sub_callback(const sensor_msgs::msg::LaserScan::SharedPtr msg)
  {
    std::cout << (msg->ranges).size() << std::endl;
    std::cout << '[';
    for (auto e : msg->ranges)
    {
      std::cout << e << ',';
    }
    std::cout << ']' << std::endl;
  }

 private:
  rclcpp::Subscription<sensor_msgs::msg::LaserScan>::SharedPtr m_sub;
};

LaserSub::LaserSub(const std::string& model): Node("topic_sub_oop_node")
{
  m_sub = this->create_subscription<sensor_msgs::msg::LaserScan>(
      model + "/scan", 10,
      std::bind(&LaserSub::sub_callback, this, std::placeholders::_1));
}

int main(int argc, char **argv)
{
  rclcpp::init(argc, argv);
  const std::string& model = std::getenv("ROBOT_MODEL");
  rclcpp::spin(std::make_shared<LaserSub>(model));
  rclcpp::shutdown();

  return 0;
}

File: python_pkg/python_pkg/laser_sub.py
# !/usr/bin/env/ python3
import rclpy
from rclpy.node import Node
from sensor_msgs.msg import LaserScan
import os

class LaserSubscriber(Node):

    def __init__(self):
        super().__init__('laser_sub_node')
        ROBOT_MODEL = os.environ['ROBOT_MODEL']
        queue_size = 10
        self.subscriber = self.create_subscription(
            LaserScan, ROBOT_MODEL+'/scan', self.subscribeCallBack, queue_size
        )
        self.subscriber  # prevent unused variable warning

    def subscribeCallBack(self, msg):
        self.get_logger().info(f'Raw Laser Data : {msg.ranges}')

def main(args=None):
    rclpy.init(args=args)

    laser_subscriber = LaserSubscriber()

    rclpy.spin(laser_subscriber)

    laser_subscriber.destroy_node()
    rclpy.shutdown()

if __name__ == '__main__':
    main()